### ISSUE TITLE
undefined method `add_filter' for nil:NilClass (NoMethodError) - when Airbrake is not configured

### ISSUE THREAD

**Issue by khustochka** [airbrake/airbrake#691]
Airbrake 5.8.0
airbrake-ruby 1.8.0

```
# config/initializers/airbrake.rb
if ENV['AIRBRAKE_PROJECT_ID'] && ENV['AIRBRAKE_API_KEY']
  Airbrake.configure do |config|
    config.project_id = ENV['AIRBRAKE_PROJECT_ID']
    config.project_key = ENV['AIRBRAKE_API_KEY']
    config.environment = Rails.env
  end
end
```

If AIRBRAKE_PROJECT_ID is not set this prevents config from being run and when starting Rails application the following error occurs:

> /Users/vkhustochka/.rvm/gems/ruby-2.4.0/gems/airbrake-5.8.0/lib/airbrake/rack/middleware.rb:40:in `block in initialize': undefined method `add_filter' for nil:NilClass (NoMethodError)

The same setup was working properly with Airbrake 5.7.1.

**Comment by kyrylo** [comment]
What's your config?

**Comment by khustochka** [comment]
Ok, I found out why this happens - because I do not load Airbrake config in development and test env, when AIRBRAKE_PROJECT_ID is not set.

```
# config/initializers/airbrake.rb
if ENV['AIRBRAKE_PROJECT_ID'] && ENV['AIRBRAKE_API_KEY']
  Airbrake.configure do |config|
    config.project_id = ENV['AIRBRAKE_PROJECT_ID']
    config.project_key = ENV['AIRBRAKE_API_KEY']
    config.environment = Rails.env
  end
end
```

This was previously discussed in the issue #546. And it was working, i.e. airbrake was not exploding if not configured.

**Comment by olbrich** [comment]
This seems to happen for me as well, if I am running tests and don't have Airbrake configured, this error is raised.

**Comment by kyrylo** [comment]
Sorry for this inconvenience and big thanks for an extremely fast bug report! I just released v5.8.1 which fixes the bug. Happy Airbraking!
